financial-connections/src/main/java/com/stripe/android/financialconnections/image/StripeImage.kt

```diff
@@ -0,0 +1,113 @@
+package com.stripe.android.financialconnections.image
+
+import android.graphics.Bitmap
+import android.util.Log
+import android.util.LruCache
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.BoxWithConstraintsScope
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.unit.Dp.Companion.Infinity
+import androidx.compose.ui.unit.IntSize.Companion.Zero
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+@Composable
+internal fun StripeImage(
+    url: String,
+    placeHolder: Painter,
+    memoryCache: LruCache<String, Bitmap> = imageMemoryCache,
+    contentDescription: String? = null,
+    modifier: Modifier = Modifier
+) {
+    BoxWithConstraints(modifier) {
+        val (width, height) = calculateBoxSize()
+        val painter: MutableState<Painter> =
+            remember { mutableStateOf(BitmapPainter(ImageBitmap(width, height))) }
+        var loadImageJob: Job?
+        val scope = rememberCoroutineScope()
+
+        DisposableEffect(url) {
```

Comment:
Should we used LaunchedEffect instead? it comes with a comes with a coroutine scope

Reply:
DisposableEffect provides a callback that helps clearing the resources (besides the actual job) used to load images:

```
onDispose {
    painter.value = BitmapPainter(ImageBitmap(width, height))
    stripeImageLoader.cancel()
    loadImageJob?.cancel()
}
```

Not sure if we can achieve that with LaunchedEffect 🤔